redundant-bmc/src/manager.cpp

```diff
@@ -42,4 +42,28 @@ sdbusplus::async::task<> Manager::doHeartBeat()
     co_return;
 }
 
+Role Manager::determineRole()
+{
+    Role role{Role::Unknown};
+
+    try
+    {
+        role_determination::Input input{
```

Comment:
I'm wondering if all the complexity we're bringing along for AIO is really needed in this app? Wont this app mostly be synchronous anyway (i.e. it can't really do anything until it determines role?). I'm just a simple person who  likes simple code. But I think this is also an opportunity to get a good grasp on what we get with the new C++ AIO features.

Reply:
You mean the coroutines?  I think they're a good improvement.  You don't have to worry about D-Bus being hung while you're making a D-Bus call to another daemon, and instead of async timer callbacks you can just use while loops.  Matches are also while loops.

Reply:
They make my head hurt though :)

Reply:
FWIW, these lines right here aren't async.  It's just a normal function.  There will eventually be a half dozen or so structure members, which is why the args aren't just passed in normally.